src/ServiceControl/MessageFailures/Handlers/MessageFailureResolvedHandler.cs

```diff
@@ -36,7 +43,8 @@ public void Handle(MessageFailureResolvedByRetry message)
 
             foreach (var alternative in message.AlternativeFailedMessageIds.Where(x => x != message.FailedMessageId))
             {
-                if (MarkMessageAsResolved(alternative))
+                if (await MarkMessageAsResolved(alternative)
```

Comment:
Same as above, WaitAll so the await isn't synchronous per message

Reply:
I deliberately left out all those kinds of optimizations. Let's first get everything working as is and then start tweaking things like that, ok?